### ISSUE TITLE
dom-repeat not workin in polymer2

### ISSUE THREAD

**Issue by jogibear9988** [Polymer/polymer#4489]
I've the following Template

    <template>
        <ul class="tile-menu" on-click="pathClick">
            <sub-tile id="moveBackTile" selectedvalue="..." on-click="moveBack"></sub-tile>
            <template is="dom-repeat" id="menuItemList" items="[[menuItems]]">
                <sub-tile style$="visibility:[[boolVisibilityConverter(item.IsSeperator)]]" id="[[item.Id]]" module="[[item.MCCModule]]" propertyname="[[item.MCCModuleItem]]" selectedvalue="[[item.Name]]" operator="[[item.Id]]" extra-parameters="[[item.ExtraParameters]]" child-items="[[item.Children]]" opennew="[[item.OpenInDialog]]"></sub-tile>
            </template>
        </ul>
    </template>

this on works in chrome, but not in firefox or safari

and I've this one

    <template>
        <ul class="tile-menu" on-click="pathClick">
            <sub-tile id="moveBackTile" selectedvalue="..." on-click="moveBack"></sub-tile>
            <dom-repeat id="menuItemList" items="[[menuItems]]">
                <sub-tile style$="visibility:[[boolVisibilityConverter(item.IsSeperator)]]" id="[[item.Id]]" module="[[item.MCCModule]]" propertyname="[[item.MCCModuleItem]]" selectedvalue="[[item.Name]]" operator="[[item.Id]]" extra-parameters="[[item.ExtraParameters]]" child-items="[[item.Children]]" opennew="[[item.OpenInDialog]]"></sub-tile>
            </dom-repeat>
        </ul>
    </template>

which does also not work on chrome!

i've added &lt;link rel="import" href="libs/polymer/lib/elements/dom-repeat.html"&gt; in my index.html !

what do i wrong? 

i use current polymer2 (rc3)!

**Comment by jsilvermist** [comment]
You need a `<template>` inside the `<dom-repeat>`.
Also, make sure dom-repeat is being imported properly if this doesn't fix the issue.

**Comment by NicolasRannou** [comment]
yes make sure to also import `dom-repeat`.

```
<link rel="import" href="../bower_components/polymer/polymer-element.html">
<link rel="import" href="../bower_components/polymer/lib/elements/dom-repeat.html">
```

**Comment by jogibear9988** [comment]
i see in ff debugger, that my array "menuItems" is set. but no elements are created!

**Comment by NicolasRannou** [comment]
what happens if you try the following code?
```
<template is="dom-repeat" id="menuItemList" items="[[menuItems]]">
<div>[[item.Id]]</div>
</template>
```

**Comment by baocang** [comment]
the docs may be out of date https://www.polymer-project.org/2.0/docs/devguide/templates.html#dom-repeat
you'll try this:
```
<template>
    <ul class="tile-menu" on-click="pathClick">
        <sub-tile id="moveBackTile" selectedvalue="..." on-click="moveBack"></sub-tile>
        <dom-repeat id="menuItemList" items="[[menuItems]]">
           <template>
              <sub-tile style$="visibility:[[boolVisibilityConverter(item.IsSeperator)]]" id="[[item.Id]]" module="[[item.MCCModule]]" propertyname="[[item.MCCModuleItem]]" selectedvalue="[[item.Name]]" operator="[[item.Id]]" extra-parameters="[[item.ExtraParameters]]" child-items="[[item.Children]]" opennew="[[item.OpenInDialog]]"></sub-tile>
           </template>
        </dom-repeat>
    </ul>
</template>
```

**Comment by jogibear9988** [comment]
if I do

    <template is="dom-repeat" id="menuItemList" items="[[menuItems]]">
    <div>[[item.Id]]</div>
    </template>

works in chrome, but not FF

**Comment by baocang** [comment]
try this:
```html
<link rel="import" href="../../bower_components/polymer/polymer-element.html">
<link rel="import" href="../../bower_components/polymer/lib/elements/dom-repeat.html">

<dom-module id="app-shell">
  <template>
    <style>
      :host {
        display: block;
      }
    </style>
    <dom-repeat items="{{items1}}">
      <template>
        <h1>OK</h1>
      </template>
    </dom-repeat>
  </template>

  <script>
    class AppShell extends Polymer.Element {
      static get is() { return 'app-shell'; }
      static get properties() {
        return {
          items1: {
            type: Array,
            value: function() {
                return [ 1,2,3,4,5 ];
            }
          }
        };
      }
    }

    window.customElements.define(AppShell.is, AppShell);
  </script>
</dom-module>
```

**Comment by jogibear9988** [comment]
when I change your code to this:

    <!DOCTYPE html>
    <script src="libs/webcomponents/webcomponents-loader.js"></script>

    <link rel="import" href="/libs/polymer/polymer-element.html">
    <link rel="import" href="/libs/polymer/lib/elements/dom-repeat.html">

    <dom-module id="app-shell">
        <template>
            <style>
                :host {
                    display: block;
                }
            </style>
            <dom-repeat items="{{items1}}">
                <template>
                    <h1>OK</h1>
                </template>
            </dom-repeat>
        </template>

        <script>
            window.addEventListener('WebComponentsReady', function (e) {
        
            class AppShell extends Polymer.Element {
                static get is() { return 'app-shell'; }
                static get properties() {
                    return {
                        items1: {
                            type: Array,
                            value: function() {
                                return [ 1,2,3,4,5 ];
                            }
                        }
                    };
                }
            }

            window.customElements.define(AppShell.is, AppShell);
            });
        </script>
    </dom-module>
    <app-shell></app-shell>

it works in FF and chrome

**Comment by jogibear9988** [comment]
and also this works:

    <!DOCTYPE html>
    <script src="libs/webcomponents/webcomponents-loader.js"></script>

    <link rel="import" href="/libs/polymer/polymer-element.html">
    <link rel="import" href="/libs/polymer/lib/elements/dom-repeat.html">

    <dom-module id="app-shell">
        <template>
            <style>
                :host {
                    display: block;
                }
            </style>
            <dom-repeat items="{{items1}}">
                <template>
                    <h1>OK</h1>
                </template>
            </dom-repeat>
        </template>

        <script>
            window.addEventListener('WebComponentsReady', function (e) {
        
                class AppShell extends Polymer.Element {
                    static get is() { return 'app-shell'; }
                    static get properties() {
                        return {
                            items1: {
                                type: Array,
                            
                            }
                        };
                    }
                }

                window.customElements.define(AppShell.is, AppShell);

                setTimeout(function () { document.getElementById('hallo').items1 = [1, 2, 3, 4, 5]; }, 3000);
            });
        </script>
    </dom-module>
    <app-shell id="hallo"></app-shell>

**Comment by jogibear9988** [comment]
i've no idea, what i'm doing wrong ;-(

**Comment by baocang** [comment]
can you shows your code? maybe your can clean your code likes this:

```html
<template>
    <!--<ul class="tile-menu" on-click="pathClick">-->
  <ul>
        <!--<sub-tile id="moveBackTile" selectedvalue="..." on-click="moveBack"></sub-tile>-->
        <dom-repeat id="menuItemList" items="[[menuItems]]">
           <template>
             <h1>OK</h1>
              <!--<sub-tile style$="visibility:[[boolVisibilityConverter(item.IsSeperator)]]" id="[[item.Id]]" module="[[item.MCCModule]]" propertyname="[[item.MCCModuleItem]]" selectedvalue="[[item.Name]]" operator="[[item.Id]]" extra-parameters="[[item.ExtraParameters]]" child-items="[[item.Children]]" opennew="[[item.OpenInDialog]]"></sub-tile>-->
           </template>
        </dom-repeat>
    </ul>
</template>
```
and check your imports, then add your code back step by step.

**Comment by sorvell** [comment]
Here's a working jsbin: http://jsbin.com/juteca/edit?html,output.

Maybe you can edit this to produce something that doesn't work on FF/Safari?

**Comment by jogibear9988** [comment]
Found my issue.... 

I used  
     
       document.getElementById('tile-menu').menudata = e.detail;

to set the data of my control instance, my instance looked like this:

     &lt;tile-menu id="tilemenu" menupath="Home" style="width: 100%; height: 100%;"%gt;

problem, instance has the same id as the dom-module with the template!

so document.getElementById returned the dom-module on FF and Safari!

i changed the instance name, and everything worked.

It would be nice if this would also fail on chrome (or produce an error)
